src/components/buttons/button.tsx

```diff
@@ -30,17 +44,23 @@ export const Button = ({ children, icon, disable, compact, type, paddingHorizont
     ...buttonStyleMapping[type],
     height: compact ? '40px' : baseButtonStyles.height,
     padding: paddingHorizontal?.toString() || baseButtonStyles.padding,
-    minWidth: expand ? '100%' : baseButtonStyles.minWidth,
+    width: expand ? '100%' : '',
   };
 
   const buttonHover = type === 'primary' && !disable ? 'primary-button' : '';
 
   return (
     <div>
       <button className={buttonHover} style={buttonStyle} disabled={disable} onClick={onClick}>
-        {!!icon && <span>{icon}</span>}
-        <Separator horizontal size={8} />
-        <label> {children} </label>
+        {!!icon && (
+          <>
+            <span>{icon}</span>
+            {icon && children && <Separator horizontal size={8} />}
+          </>
+        )}
+        <LargeLabel weight={weight} color={color}>
+          {children}
+        </LargeLabel>
```

Comment:
aqui mudaria apenas uma coisa:
```suggestion
        <LargeLabel weight={weight} color={color}>
          {title}
        </LargeLabel>
```